be/src/http/action/compaction_action.h

```diff
@@ -41,8 +52,25 @@ class CompactionAction : public HttpHandler {
 private:
     Status _handle_show_compaction(HttpRequest *req, std::string* json_result);
 
+    /// execute compaction request to run compaction task
+    /// param compact_type in req to distinguish the task type, base or cumulative
+    Status _handle_run_compaction(HttpRequest *req, std::string* json_result);
+
+    /// thread callback function for the tablet to do compaction
+    OLAPStatus _execute_compaction_callback(TabletSharedPtr tablet, const std::string& compaction_type);
+
+    /// fetch compaction running status
+    Status _handle_run_status_compaction(HttpRequest* req, std::string* json_result);
+
 private:
     CompactionActionType _type;
+
+    /// running check mutex
+    static std::mutex _compaction_running_mutex;
+    /// whether there is manual compaction running
+    static bool _is_compaction_running;
```

Comment:
Is an atomic\<bool> OK?

Reply:
already use mutex

Reply:
I mean, mutex is not needed, you can use atomic<bool> instead, I found `_compaction_running_mutex ` is only aim to protect `_is_compaction_running `

Reply:
Only use atomic<bool> can not cover this scenes, becouse atomic<bool> do not have test and set operation.